File: Misc/kytdkut_Import all media in project directory - query padding percent.py
# description Import all media in project directory - query padding percent
# @version 1.0
# @author kytdkut

# Sound in Words - only intended for internal use
# this is useful when creating RPP files via scripting
# this script imports all the wav media contained in the media_path folder and separates it using the prompted value (kind of)

import reaper_python
from pathlib import Path
from sys import exit

media_files = []
user_input = RPR_GetUserInputs("Insert padding amount", 1, "LoSy percent amount:", 10, 100)
padding_percent = float(user_input[4]) / 100

(path, len) = RPR_GetProjectPath("", 512)
media_path = Path(path)

for file in media_path.glob("[!$._]*.wav"):
    media_files.append(file)

media_files = sorted(media_files)


def main():
    RPR_Undo_BeginBlock()
    
    for file in media_files:
        RPR_InsertMedia(file, 0)
    
    selected_track = RPR_GetSelectedTrack(0, 0)
    num_items = RPR_CountTrackMediaItems(selected_track)
    item_durs = []
    
    
    for i in range(num_items): # get item lengths
        current_item = RPR_GetTrackMediaItem(selected_track, i)
        item_durs.append(RPR_GetMediaItemInfo_Value(current_item, 'D_LENGTH'))

    for i in reversed(range(num_items)): # reposition items leaving 2 times its length in silence afterwards
        current_item = RPR_GetTrackMediaItem(selected_track, i)
        current_item_pos = RPR_GetMediaItemInfo_Value(current_item, 'D_POSITION')
        RPR_SetMediaItemInfo_Value(current_item, 'D_POSITION', current_item_pos * 3)

    item_durs_cut = item_durs # for safety?
    item_durs_cut.pop() # removing last item because we don't need its duration

    for i in reversed(range(num_items)): # get padding_percent of all previous_item_durs and add to current_item_pos
        current_item = RPR_GetTrackMediaItem(selected_track, i)
        current_item_pos = RPR_GetMediaItemInfo_Value(current_item, 'D_POSITION')
        previous_item_durs = sum(item_durs_cut)
        RPR_SetMediaItemInfo_Value(current_item, 'D_POSITION', current_item_pos + previous_item_durs * padding_percent)
        if i > 0: # this is ugly, but the list does not have any more elements to pop now
            item_durs_cut.pop() # removing last element (item dur) each loop

    for i in reversed(range(num_items)): # add 2 extra seconds between items
        current_item = RPR_GetTrackMediaItem(selected_track, i)
        current_item_pos = RPR_GetMediaItemInfo_Value(current_item, 'D_POSITION')
        RPR_SetMediaItemInfo_Value(current_item, 'D_POSITION', current_item_pos + 2 * i)

    RPR_Undo_EndBlock("Import all media in project directory", -1)


if user_input[0] == 1:
    RPR_SetEditCurPos(0, 1, 1)
    main()
    RPR_SetEditCurPos(0, 1, 1)

File: Misc/kytdkut_Import all media in project directory - query padding percent.lua
-- description Import all media in project directory - query padding percent
-- @version 1.0
-- @author kytdkut

-- Sound in Words - only intended for internal use
local project_path = reaper.GetProjectPath(0)
local files = {}
local item_durs = {}
local retval, padding_percent = reaper.GetUserInputs("Insert padding amount", 1, "LoSy percent amount:", 10, 100)

function GetFilesInMediaFolder()
    local i = 0
    repeat
        local file = reaper.EnumerateFiles(project_path, i)
        if file then
            local ext = file:match('.-%.(.*)')
            if reaper.IsMediaExtension(ext, false) then
                files[i + 1] = file -- arrays in lua start on 1
            end
            i = i + 1
        end
    until not file
end

function RepositionItems()
    local selected_track = reaper.GetSelectedTrack(0, 0)
    local num_items = reaper.CountTrackMediaItems(selected_track)
    -- get durs
    for i = 0, num_items - 1 do
        local current_item = reaper.GetTrackMediaItem(selected_track, i)
        item_durs[i + 1] = reaper.GetMediaItemInfo_Value(current_item, 'D_LENGTH')
    end
    -- first repositioning - 2x item dur
    for i = num_items - 1, 1, -1 do -- reverse iterate
        local current_item = reaper.GetTrackMediaItem(selected_track, i)
        local current_item_pos = reaper.GetMediaItemInfo_Value(current_item, 'D_POSITION')
        reaper.SetMediaItemInfo_Value(current_item, 'D_POSITION', current_item_pos * 3)
    end
    -- second repositioning
    local item_durs_sum = 0
    for _, dur in ipairs(item_durs) do
        item_durs_sum = item_durs_sum + dur
    end
    for i = num_items - 1, 1, -1 do
        local current_item = reaper.GetTrackMediaItem(selected_track, i)
        local current_item_pos = reaper.GetMediaItemInfo_Value(current_item, 'D_POSITION')
        local current_item_dur = reaper.GetMediaItemInfo_Value(current_item, 'D_LENGTH')
        item_durs_sum = item_durs_sum - current_item_dur -- discard current item dur
        reaper.SetMediaItemInfo_Value(current_item, 'D_POSITION', current_item_pos + item_durs_sum * padding_percent / 100)
    end
    -- third repositioning - add 2 secs
    for i = num_items - 1, 1, -1 do
        local current_item = reaper.GetTrackMediaItem(selected_track, i)
        local current_item_pos = reaper.GetMediaItemInfo_Value(current_item, 'D_POSITION')
        reaper.SetMediaItemInfo_Value(current_item, 'D_POSITION', current_item_pos + 2 * i)
    end
end

if retval then
    reaper.Undo_BeginBlock()
    reaper.SetEditCurPos(0, 1, 1)
    GetFilesInMediaFolder()
    for i in pairs(files) do
        reaper.InsertMedia(project_path .. "/"  .. files[i], 0)
    end
    RepositionItems()
    reaper.SetEditCurPos(0, 1, 1)
    reaper.Undo_EndBlock("Import all media in project directory - query padding percent", -1)
end
